src/collection.js

```diff
@@ -626,13 +629,27 @@ export default class Collection {
     options = Object.assign({
       strategy: Collection.strategy.MANUAL,
       lastModified: this.lastModified,
+      limit: options.fetchLimit || null,
+      token: options.fetchToken || null,
       headers: {},
     }, options);
     // First fetch remote changes from the server
     return this.api.fetchChangesSince(this.bucket, this.name, {
       lastModified: options.lastModified,
+      limit: options.limit,
+      token: options.token,
       headers: options.headers
     })
+      // Deal with pagination
+      .then(changes => {
+        if (changes.fetchToken) {
+          syncResultObject.fetchToken = changes.fetchToken;
+        }
+        // TODO: deal with the case where records disappear from
+        // pages already fetched, due to updates to the server
+        // while the paginating query series is in progress.
```

Comment:
There are two strategies here.
The first one is to send the ``If-Match`` header and expect a 412 in case the page fails. The problem with this strategy is that you cannot really now how long you will have to restart the process in case the collection change a lot.

The second strategy (and the one I think we should take for Kinto.js) is not to send the If-Match header and to follow the pagination til this end.

And then ask the server all the changes that happened since the beginning of the fetching (taking the Etag from the first page of the fetching)